hack/build-go.sh

```diff
@@ -126,16 +126,27 @@ for arg in $args; do
             kubevirt::version::get_version_vars
             echo "$KUBEVIRT_GIT_VERSION" >${CMD_OUT_DIR}/${BIN_NAME}/.version
 
-            # build virtctl also for darwin and windows and for arm64 when on amd64 on linux
-            if [ "${BIN_NAME}" = "virtctl" -a "${ARCH}" = "amd64" -a -z "${LINUX_ONLY}" ]; then
+            # build virtctl for all architectures if requested
+            if [ "${BIN_NAME}" = "virtctl" -a "${VIRTCTL_BUILD_ALL_ARCH}" = "true" ]; then
+                GOOS=linux GOARCH=amd64 go_build -tags "${KUBEVIRT_GO_BUILD_TAGS}" -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-linux-amd64 -ldflags "$(kubevirt::version::ldflags)" $(pkg_dir linux amd64)
```

Comment:
Wouldn't it be better to have two loops instead of copy-pasting the build cmd for each arch/platfrom combination? E.g.
```
foreach {amd64, arm64}
  foreach {linux, darwin, windows}
    GOOS=....
```

Reply:
Nice catch! I think this is much better! Thanks